File: PYTHON/ProgramStudi.py
# # import class parent
from Course import Course


class ProgramStudi:
    # constructor
    def __init__(self, nama_prodi, kode, courses=None):
        self._nama_prodi = nama_prodi
        self._kode = kode
        # inisialisais array untuk course
        if courses is None:
            self._courses = []
        else:
            self._courses = courses

    # getter dan setter
    def getNamaProdi(self):
        return self._nama_prodi

    def setNamaProdi(self, nama_prodi):
        self._nama_prodi = nama_prodi

    def getKode(self):
        return self._kode

    def setKode(self, kode):
        self._kode = kode

    def get_courses(self):
        return self._courses

    # menambah dan menghapus courses yang diambil oleh programstudi
    def tambah_course(self, course):
        self._courses.append(course)

    def hapus_course(self, course):
        self._courses.remove(course)

File: C++/ProgramStudi.cpp
#include "ProgramStudi.h"

// constructor
ProgramStudi::ProgramStudi(std::string nama_prodi, std::string kode, std::list<Course> courses)
    : _nama_prodi(nama_prodi), _kode(kode), _courses(courses) {}

// getter dan setter
std::string ProgramStudi::getNamaProdi() {
    return _nama_prodi;
}

void ProgramStudi::setNamaProdi(std::string nama_prodi) {
    _nama_prodi = nama_prodi;
}

std::string ProgramStudi::getKode() {
    return _kode;
}

void ProgramStudi::setKode(std::string kode) {
    _kode = kode;
}

std::list<Course> ProgramStudi::getCourses() {
    return _courses;
}

void ProgramStudi::setCourses(std::list<Course> courses) {
    _courses = courses;
}

// menambah dan menghapus courses yang diambil oleh programstudi
void ProgramStudi::tambahCourse(Course course) {
    _courses.push_back(course);
}

void ProgramStudi::hapusCourse(Course course) {
    _courses.remove(course);
}
